https://bugreports.qt.io/browse/QTBUG-124346

From 36a1fb7256fb4c6857d913ebb2beb4096fde3bad Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Fri, 12 Apr 2024 16:12:38 +0200
Subject: [PATCH] Expose the qversiontagging.h header file unconditionally

The header file is not guarded properly in qglobal.h. This installs
it unconditionally in systems where version tagging is not supported.

Task-number: QTBUG-124346
Change-Id: I130b7d016e28a7a03d8929bc17fad50c80648f49
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
---
 src/corelib/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/CMakeLists.txt b/src/corelib/CMakeLists.txt
index 8c2f62ef03b8..a785a31ee3e0 100644
--- qt-everywhere-src-6.7.0/qtbase/src/corelib/CMakeLists.txt
+++ qt-everywhere-src-6.7.0/qtbase/src/corelib/CMakeLists.txt
@@ -92,6 +92,7 @@ qt_internal_add_module(Core
         global/q23utility.h
         global/qxpfunctional.h
         global/qxptype_traits.h
+        global/qversiontagging.h
         ipc/qsharedmemory.cpp ipc/qsharedmemory.h ipc/qsharedmemory_p.h
         ipc/qsystemsemaphore.cpp ipc/qsystemsemaphore.h ipc/qsystemsemaphore_p.h
         ipc/qtipccommon.cpp ipc/qtipccommon.h ipc/qtipccommon_p.h
@@ -453,8 +454,7 @@ endif()
 # Add version tagging source files if the linker has version script support
 # or the platform supports it.
 set(core_version_tagging_files
-    global/qversiontagging.cpp
-    global/qversiontagging.h)
+    global/qversiontagging.cpp)
 qt_internal_extend_target(Core
     CONDITION TEST_ld_version_script OR APPLE OR WIN32
     SOURCES ${core_version_tagging_files}
-- 
2.16.3

From fc753620bcbc24cfa469f9b68ef649201240fc35 Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Tue, 23 Apr 2024 14:51:57 +0200
Subject: [PATCH] Define QT_NO_VERSION_TAGGING for platforms that do not
 support version tagging

Task-number: QTBUG-124346
Change-Id: I210f61d5748d2def24f6bfe0d58fd82040c6735c
---
 src/corelib/CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/corelib/CMakeLists.txt b/src/corelib/CMakeLists.txt
index e1fb0d7e393e..830f3e2503fe 100644
--- qt-everywhere-src-6.7.0/qtbase/src/corelib/CMakeLists.txt
+++ qt-everywhere-src-6.7.0/qtbase/src/corelib/CMakeLists.txt
@@ -459,7 +459,12 @@ qt_internal_extend_target(Core
     CONDITION TEST_ld_version_script OR APPLE OR WIN32
     SOURCES ${core_version_tagging_files}
 )
-
+# Define QT_NO_VERSION_TAGGING otherwise
+qt_internal_extend_target(Core
+    CONDITION NOT TEST_ld_version_script AND NOT APPLE AND NOT WIN32
+    PUBLIC_DEFINES
+        QT_NO_VERSION_TAGGING
+)
 
 if(GCC)
     # Disable LTO, as the symbols disappear somehow under GCC
-- 
2.16.3

